Mathlib/RingTheory/AdicCompletion/LocalRing.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2025 Nailin Guan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nailin Guan
+-/
+
+import Mathlib.RingTheory.AdicCompletion.Basic
+import Mathlib.RingTheory.LocalRing.Defs
+
+/-!
+# Basic Properties of Complete Local Ring
+
+In this file we proved that a ring that is adic complete with respect to a maximal ideal
+ia a local ring (complete local ring).
+
+-/
+
+variable {R : Type*} [CommRing R] (m : Ideal R) [hmax : m.IsMaximal]
+
+open Ideal Quotient
+
+lemma isUnit_iff_nmem_of_isAdicComplete_maximal [IsAdicComplete m R] (r : R) :
+    IsUnit r ↔ r ∉ m := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · by_contra mem
+    rcases IsUnit.exists_left_inv h with ⟨s, hs⟩
+    absurd m.ne_top_iff_one.mp (Ideal.IsMaximal.ne_top hmax)
+    simp [← hs, Ideal.mul_mem_left m s mem]
+  · have mapu {n : ℕ} (npos : n > 0) : IsUnit (Ideal.Quotient.mk (m ^ n) r) := by
```

Comment:
Please use `0 < n` throughout

Reply:
Wait, didn't we have a linter for this?!